lib/adagrams.rb

```diff
@@ -0,0 +1,188 @@
+require 'pry'
+# Wave 1
+
+def draw_letters
+  quantity_by_letter = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+  }
+  
+  # Put quantity of each bank of letters in an arry
+  array_letters = quantity_by_letter.map do |letter, quantity|
+    letter.to_s * quantity
+  end
+  
+  array_letters = array_letters.join("")
+  array_letters = array_letters.split('')
+  
+  # Create hand for the user
+  array_hand = array_letters.sample(10)
+  
+  return array_hand
+  
+end
+
+# Wave 2
+# A method that will return true if user enters same letters in hand, else false
+def uses_available_letters?(user_input_word, letters_in_hand)
+  # Create array of user input - make uppercase
+  user_input_array = user_input_word.upcase.split("")
+  
+  hand_hash = Hash.new(0)
+  input_hash = Hash.new(0)
+  
+  # Hash from hand - quantity of individual letters
+  letters_in_hand.each do |character|
+    hand_hash[character] += 1
+  end
```

Comment:
Once you have this hash, do you need a second one? Or can you just iterate through the input and reduce the values on the `hand_hash`?